app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -131,4 +176,78 @@ private void expandSelected(boolean expand) {
             selectedCollected.findViewById(R.id.collected_date).setVisibility(expand ? View.VISIBLE:View.GONE);
         }
     }
+
+    /**
+     * On sign out button click
+     */
+    public void signOutButton(View view) {
+        AuthService.getInstance().signOut(this);
+        finish();
+    }
+
+    /**
+     * On change username button click
+     */
+    public void changeUsernameButton(View view) {
+        // Hide username
+        findViewById(R.id.profile_username).setVisibility(View.GONE);
+
+        // Show edit text
+        findViewById(R.id.profile_username_edit).setVisibility(View.VISIBLE);
+
+        // Change listener so next click will result in a username chance
+        findViewById(R.id.profile_change_username).setOnClickListener(this::confirmUsernameButton);
+    }
+
+    /**
+     * On confirm username change button click
+     */
+    public void confirmUsernameButton(View view) {
```

Comment:
I tested the change username functionality. 
Is it possible to change a bit of how it works. 

Currently when you press the change username button

1. It changes to an edit text view that looks different from the normal text view (other colors smaller text etc.).
2. You press the new text view to type in your name
3. You press the change username button again to do the change.

Would it be possible to change this flow to, when you press the button:

1. The view changes to edit text (that looks just like the username text, and you get the keyboard directly, to type in a new username. 
2. When you press enter on the keyboard it changes name, if you press anywhere else on the screen it stops trying to change username? 

I think that flow would make more sense and it removes 2 extra clicks for the user

Reply:
The color is actually the same for me, but I agree that it would be nice for the keyboard to appear when clicking the first time on the button and to have an option to cancel the username change (like pressing anywhere else on the screen). I think that this is a "nice to have" so maybe you shouldn't sacrifice too much time on this right now if you can't find a fast way to implement these features.

Reply:
Suggested some changes below to handle these points pointed out by Alex.

Reply:
Thanks to you both for the suggestion (and for the solution @magangloff !).
Now, when you press on the username change button, the edit text and the keyboard are shown, whereas the button and the username text view are set as gone. 
When you press enter on the keyboard the change is performed (or denied, if necessary). 

Also, with the solution proposed by @magangloff we solved also the problem of how to force a username change for new users (in other words, how to handle the registration phase). 
Basically, we force new users to insert something in the edit text and even if you try to press somewhere else on the screen, nothing changes.

You can test again the functionality and tell me if now it is good for you.